frontend/src/features/public-form/components/StatusTrackerPage/TimelineRunSteps.tsx

```diff
@@ -0,0 +1,137 @@
+import { MdCheck, MdClose } from 'react-icons/md'
+import { Box, Circle, Divider, Flex, Icon, Stack, Text } from '@chakra-ui/react'
+import { format } from 'date-fns'
+
+import { StepData, WorkflowStatus } from '~shared/types'
+
+const statusColor: Record<WorkflowStatus, string> = {
```

Comment:
Could be tighter for the CSS color values.
```suggestion
const statusColor: Record<WorkflowStatus, BackgroundProps['bg']> = {
```